src/main/kotlin/server/TrackerQueryExecutor.kt

```diff
@@ -0,0 +1,106 @@
+package server
+
+import com.intellij.openapi.application.PathManager
+import models.Extension
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.MultipartBody
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.asRequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import java.io.File
+import java.net.URL
+
+class TrackerQueryExecutor : QueryExecutor() {
+
+    companion object {
+        private val ACTIVITY_TRACKER_FILE = "ide-events${Extension.CSV.ext}"
+        private const val CODE_TRACKER_FILE_FIELD = "codetracker"
+        private const val ACTIVITY_TRACKER_FILE_FIELD = "activitytracker"
+        private val activityTrackerPath = "${PathManager.getPluginsPath()}/activity-tracker/" + ACTIVITY_TRACKER_FILE
+    }
+
+    private var activityTrackerKey: String? = null
+
+    init {
+        initActivityTrackerInfo()
+    }
+
+    private fun initActivityTrackerInfo() {
+        val currentUrl = URL(baseUrl + "activity-tracker-item")
+        logger.info("${Plugin.PLUGIN_ID}: ...generating activity tracker key")
+
+        val request = Request.Builder().url(currentUrl).post(
```

Comment:
Hm, you are sending empty request?

Reply:
Yes, it is necessary to create an activity-tracker file and send id for if